engine/src/main/java/org/pentaho/di/trans/steps/pgbulkloader/PGBulkLoader.java

```diff
@@ -133,12 +136,40 @@ public String getCopyCommand( ) throws KettleException {
     return contents.toString();
   }
 
+  void checkClientEncoding() throws Exception {
+    Connection connection = data.db.getConnection();
+
+    Statement statement = connection.createStatement();
+
+    try {
+      try (ResultSet rs = statement.executeQuery("show client_encoding")) {
+        if (!rs.next()) {
+          throw new KettleException( "Cannot detect client encoding" );
```

Comment:
It should not be an exception that the client encoding cannot be auto-detected. Rather, you can log a warning, and default the client encoding to the system default encoding. If for some reason you cannot get the encoding, existing systems should continue to work as they have before.